test/src/unit-max-tile-size.cc

```diff
@@ -0,0 +1,1136 @@
+        tiledb::Attribute::create<std::array<char, 42>>(ctx_, "c42");
+    schemaEvolution.add_attribute(c);
+
+    schemaEvolution.drop_attribute(std::string("b257"));
+
+    // evolve array
+    schemaEvolution.array_evolve(array_name);
+  };
+
+  remove_temp_dir(array_name);
+
+  SECTION(" - Create") {
+    create_dense_1d_d1_var_a1_fix<int, std::array<char, 2>>(
+        array_name, { 1, 4 }, 1, "a2");
+    CHECK(c_get_fragments_max_in_memory_tile_size(array_name) == 0);
+    write_array_attr<2>(array_name, "a2");
+    CHECK(c_get_fragments_max_in_memory_tile_size(array_name) == 4);
+
+    remove_temp_dir(array_name);
+    create_dense_1d_d1_var_a1_fix<int, std::array<char, 257>>(
+        array_name, {1, 4}, 1, "b257");
+    CHECK(c_get_fragments_max_in_memory_tile_size(array_name) == 0);
+    write_array_attr<257>(array_name, "b257");
+    CHECK(c_get_fragments_max_in_memory_tile_size(array_name) == 257);
+
+    remove_temp_dir(array_name);
+    create_dense_1d_d1_var_a1_fix<int, std::array<char, 42>>(
+        array_name, {1, 4}, 1, "c42");
+    CHECK(c_get_fragments_max_in_memory_tile_size(array_name) == 0);
+    write_array_attr<42>(array_name, "c42");
+    CHECK(c_get_fragments_max_in_memory_tile_size(array_name) == 42);
+  }
+  SECTION(" - Evolve") {
+    create_dense_1d_d1_var_a1_fix<int, std::array<char, 2>>(
+        array_name, {1, 4}, 1, "a2");
+    CHECK(c_get_fragments_max_in_memory_tile_size(array_name) == 0);
+    write_array_attr<2>(array_name, "a2");
+    CHECK(c_get_fragments_max_in_memory_tile_size(array_name) == 4);
+
+    array_schema_evolve_A();
+    CHECK(c_get_fragments_max_in_memory_tile_size(array_name) == 4);
+    write_array_attr<257>(array_name, "b257");
+    CHECK(c_get_fragments_max_in_memory_tile_size(array_name) == 257);
+
+    array_schema_evolve_B();
+    CHECK(c_get_fragments_max_in_memory_tile_size(array_name) == 257);
+    write_array_attr<42>(array_name, "c42");
+    // earlier fragment should still have value of 257
+    CHECK(c_get_fragments_max_in_memory_tile_size(array_name) == 257);
+
+    // now want to consolidate, but not vacuum, max
+    // should still be 257
+    tiledb::Array::consolidate(ctx_, array_name);
+    // After consolidation, old fragment should still be there with 257
+    CHECK(c_get_fragments_max_in_memory_tile_size(array_name) == 257);
+
+    // (sparse/dense vacuum/consolidate semantics differ, 
+    //  vacuum dense eliminates old data)
+    // then vacuum, now max should become 42 ...
+    tiledb::Array::vacuum(ctx_, array_name);
+    CHECK(c_get_fragments_max_in_memory_tile_size(array_name) == 42);
+  }
+}
+
+TEST_CASE_METHOD(
+    MaxTileSizeFx,
+    "C++ API: Max fragment size, sparse, variable string dimension, variable string attribute",
+    "[capi][max-tile-size][sparse][evolve]") {
+  std::string array_name("tile_size_array");
+
+  remove_temp_dir(array_name);
+
+  create_sparse_1d_d1_var_a1_var<std::string>(
+      array_name,
+      true     // bool is_nullable
```

Comment:
reasoning was doing one to show what the not-necessarily-obvious parameters were, since the data by sight alone can be easily confused (at least for me.)

Reply:
Let's take it out, most IDEs are good at showing this information when needed.